### ISSUE TITLE
called `Result::unwrap()` on an `Err` value: HierarchyRequest

### ISSUE THREAD

**Issue by mateon1** [servo/servo#18436]
Found when throwing Servo at top alexa sites
Affected URLs:
```
http://doramatv.ru
http://readmanga.me
http://radikal.ru
http://serviporno.com
```

Debug stack trace (radikal.ru):
```gdb
called `Result::unwrap()` on an `Err` value: HierarchyRequest (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(9) }, at /checkout/src/libcore/result.rs:906)
stack backtrace:
   0:     0x560a0bfab2e4 - backtrace::backtrace::libunwind::trace
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x560a0bfa586f - backtrace::capture::{{impl}}::new
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x560a076d8ccc - servo::main::{{closure}}
                        at /shared/dev/rust/servo/ports/servo/main.rs:130
   3:     0x560a0e9bf316 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:612
   4:     0x560a0e9bf134 - std::panicking::begin_panic<alloc::string::String>
                        at /checkout/src/libstd/panicking.rs:572
   5:     0x560a0e9bf0a9 - std::panicking::begin_panic_fmt
                        at /checkout/src/libstd/panicking.rs:522
   6:     0x560a0e9bf03a - std::panicking::rust_begin_panic
                        at /checkout/src/libstd/panicking.rs:498
   7:     0x560a0e9f9660 - core::panicking::panic_fmt
                        at /checkout/src/libcore/panicking.rs:71
   8:     0x560a0984ba65 - core::result::unwrap_failed<script::dom::bindings::error::Error>
                        at /checkout/src/libcore/macros.rs:41
   9:     0x560a0af22cff - core::result::{{impl}}::unwrap<script::dom::bindings::js::Root<script::dom::node::Node>,script::dom::bindings::error::Error>
                        at /checkout/src/libcore/result.rs:772
  10:     0x560a0b23cae8 - script::dom::servoparser::insert
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:723
  11:     0x560a0b23e9f7 - script::dom::servoparser::{{impl}}::append
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:870
  12:     0x560a0999d32b - html5ever::tree_builder::{{impl}}::create_root<script::dom::bindings::js::JS<script::dom::node::Node>,script::dom::servoparser::Sink>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tree_builder/mod.rs:1162
  13:     0x560a099def8f - html5ever::tree_builder::{{impl}}::step<script::dom::bindings::js::JS<script::dom::node::Node>,script::dom::servoparser::Sink>
                        at /shared/dev/rust/servo/target/debug/build/html5ever-460a31fab6c90744/out/rules.rs:42
  14:     0x560a099bbf75 - html5ever::tree_builder::{{impl}}::process_to_completion<script::dom::bindings::js::JS<script::dom::node::Node>,script::dom::servoparser::Sink>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tree_builder/mod.rs:313
  15:     0x560a0947e459 - html5ever::tree_builder::{{impl}}::process_token<script::dom::bindings::js::JS<script::dom::node::Node>,script::dom::servoparser::Sink>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tree_builder/mod.rs:475
  16:     0x560a0afd81db - html5ever::tokenizer::{{impl}}::process_token<html5ever::tree_builder::TreeBuilder<script::dom::bindings::js::JS<script::dom::node::Node>, script::dom::servoparser::Sink>>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:233
  17:     0x560a0afdcaec - html5ever::tokenizer::{{impl}}::process_token_and_continue<html5ever::tree_builder::TreeBuilder<script::dom::bindings::js::JS<script::dom::node::Node>, script::dom::servoparser::Sink>>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:238
  18:     0x560a0afd6346 - html5ever::tokenizer::{{impl}}::emit_chars<html5ever::tree_builder::TreeBuilder<script::dom::bindings::js::JS<script::dom::node::Node>, script::dom::servoparser::Sink>>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:397
  19:     0x560a0afe6fe6 - html5ever::tokenizer::{{impl}}::step<html5ever::tree_builder::TreeBuilder<script::dom::bindings::js::JS<script::dom::node::Node>, script::dom::servoparser::Sink>>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:688
  20:     0x560a0afdf316 - html5ever::tokenizer::{{impl}}::run<html5ever::tree_builder::TreeBuilder<script::dom::bindings::js::JS<script::dom::node::Node>, script::dom::servoparser::Sink>>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:362
  21:     0x560a0afdf5f2 - html5ever::tokenizer::{{impl}}::feed<html5ever::tree_builder::TreeBuilder<script::dom::bindings::js::JS<script::dom::node::Node>, script::dom::servoparser::Sink>>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:220
  22:     0x560a0a6c22da - script::dom::servoparser::html::{{impl}}::feed
                        at /shared/dev/rust/servo/components/script/dom/servoparser/html.rs:79
  23:     0x560a0b2397b9 - script::dom::servoparser::{{impl}}::feed
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:509
  24:     0x560a0b238507 - script::dom::servoparser::{{impl}}::do_parse_sync::{{closure}}
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:395
  25:     0x560a0b238e0c - script::dom::servoparser::{{impl}}::tokenize<closure>
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:432
  26:     0x560a0b2382ff - script::dom::servoparser::{{impl}}::do_parse_sync
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:395
  27:     0x560a0b2380c7 - script::dom::servoparser::{{impl}}::parse_sync::{{closure}}
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:381
  28:     0x560a0a7ee0ec - profile_traits::time::profile<(),closure>
                        at /shared/dev/rust/servo/components/profile_traits/time.rs:124
  29:     0x560a0b237fb7 - script::dom::servoparser::{{impl}}::parse_sync
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:378
  30:     0x560a0b23864c - script::dom::servoparser::{{impl}}::parse_string_chunk
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:412
  31:     0x560a0b2347d6 - script::dom::servoparser::{{impl}}::parse_html_document
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:118
  32:     0x560a09e7ca44 - script::script_thread::{{impl}}::load
                        at /shared/dev/rust/servo/components/script/script_thread.rs:2164
  33:     0x560a09e74ae8 - script::script_thread::{{impl}}::handle_page_headers_available::{{closure}}
                        at /shared/dev/rust/servo/components/script/script_thread.rs:1687
  34:     0x560a09569b9e - core::option::{{impl}}::map<net_traits::Metadata,script::dom::bindings::js::Root<script::dom::servoparser::ServoParser>,closure>
                        at /checkout/src/libcore/option.rs:398
  35:     0x560a09e74780 - script::script_thread::{{impl}}::handle_page_headers_available
                        at /shared/dev/rust/servo/components/script/script_thread.rs:1687
  36:     0x560a0b285feb - script::script_thread::{{impl}}::page_headers_available::{{closure}}
                        at /shared/dev/rust/servo/components/script/script_thread.rs:692
  37:     0x560a0ac52604 - std::thread::local::{{impl}}::try_with<core::cell::Cell<core::option::Option<*const script::script_thread::ScriptThread>>,closure,core::option::Option<script::dom::bindings::js::Root<script::dom::servoparser::ServoParser>>>
                        at /checkout/src/libstd/thread/local.rs:365
  38:     0x560a0ab36bfc - std::thread::local::{{impl}}::with<core::cell::Cell<core::option::Option<*const script::script_thread::ScriptThread>>,closure,core::option::Option<script::dom::bindings::js::Root<script::dom::servoparser::ServoParser>>>
                        at /checkout/src/libstd/thread/local.rs:279
  39:     0x560a09e5f77a - script::script_thread::{{impl}}::page_headers_available
                        at /shared/dev/rust/servo/components/script/script_thread.rs:690
  40:     0x560a0b23a1fb - script::dom::servoparser::{{impl}}::process_response
                        at /shared/dev/rust/servo/components/script/dom/servoparser/mod.rs:605
  41:     0x560a09e82782 - script::script_thread::{{impl}}::handle_fetch_metadata
                        at /shared/dev/rust/servo/components/script/script_thread.rs:2419
  42:     0x560a09e6936a - script::script_thread::{{impl}}::handle_msg_from_constellation
                        at /shared/dev/rust/servo/components/script/script_thread.rs:1216
  43:     0x560a09e6684e - script::script_thread::{{impl}}::handle_msgs::{{closure}}
                        at /shared/dev/rust/servo/components/script/script_thread.rs:1096
  44:     0x560a09e67ad0 - script::script_thread::{{impl}}::profile_event<closure,core::option::Option<bool>>
                        at /shared/dev/rust/servo/components/script/script_thread.rs:1208
  45:     0x560a09e63d72 - script::script_thread::{{impl}}::handle_msgs
                        at /shared/dev/rust/servo/components/script/script_thread.rs:1090
  46:     0x560a09e61b45 - script::script_thread::{{impl}}::start
                        at /shared/dev/rust/servo/components/script/script_thread.rs:925
  47:     0x560a09e5f374 - script::script_thread::{{impl}}::create::{{closure}}::{{closure}}
                        at /shared/dev/rust/servo/components/script/script_thread.rs:618
  48:     0x560a09490e8e - profile_traits::mem::{{impl}}::run_with_memory_reporting<closure,fn(profile_traits::mem::ReportsChan) -> script::script_runtime::CommonScriptMsg,script::script_runtime::CommonScriptMsg,std::sync::mpsc::Sender<script::script_thread::MainThreadScriptMsg>>
                        at /shared/dev/rust/servo/components/profile_traits/mem.rs:63
  49:     0x560a0b28546b - script::script_thread::{{impl}}::create::{{closure}}
                        at /shared/dev/rust/servo/components/script/script_thread.rs:617
  50:     0x560a0aaff82a - std::sys_common::backtrace::__rust_begin_short_backtrace<closure,()>
                        at /checkout/src/libstd/sys_common/backtrace.rs:136
  51:     0x560a0b65a2d3 - std::thread::{{impl}}::spawn::{{closure}}::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:394
  52:     0x560a0b38a4ca - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  53:     0x560a0a2c20f2 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  54:     0x560a0e9c63bc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  55:     0x560a0a15d5cc - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  56:     0x560a0b6100e5 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  57:     0x560a0ab0a19e - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:393
  58:     0x560a0afd1c13 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /checkout/src/liballoc/boxed.rs:682
  59:     0x560a0e9bdf9b - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/liballoc/boxed.rs:692
                         - std::sys_common::thread::start_thread
                        at /checkout/src/libstd/sys_common/thread.rs:21
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:84
  60:     0x7fd7b42fa493 - start_thread
  61:     0x7fd7b3e27abe - __clone
  62:                0x0 - <unknown>
ERROR:servo: called `Result::unwrap()` on an `Err` value: HierarchyRequest
```

**Comment by teymour-aldridge** [comment]
I cannot reproduce this - I think the defect has been fixed.

**Comment by mateon1** [comment]
I tried to triage this with saved samples I have, but I can't seem to get a recent version of servo running, as they segfault or panic in graphics bindings, even in headless mode. This would be a good issue to debug over IRC if that still existed, not sure where servo devs gather nowadays.

I have a couple reduced samples that used to trigger the same panic, with a slightly different stack trace (Caused by JS DOM modification, not HTML parsing), and other unreduced samples that are too big to share.

Unfortunately I believe the original bug in this issue may have depended on network behavior or something like misbehaving ads. In hindsight I wish I captured some WARCs that could be replayed.

I think it's safe to close this issue and reopen, or open another if this reoccurs.
